test/library/standard/Heap/type/HeapTest.chpl

```diff
@@ -33,37 +33,39 @@ proc testHeap(type t) where isTuple(t) {
   assert(l.size == 0);
 }
 
-proc testHeap(type t) {
+proc testHeap(type t) where isBorrowedClass(t) {
   var l = new heap(t);
 
-  var x: t = new t(1);
+  // create values with 'owned' if t is a borrowed class type
+  // (the set will still store borrowed)
+  type useT = (t:owned);
 
-  l.push(x);
+  var x: useT = new useT(1);
+
+  l.push(x.borrow());
   assert(l.size == 1);
 
   var value = l.top();
   l.pop();
   assert(l.size == 0);
-
-  if isUnmanagedClass(t) {
-    delete x;
-  }
-
 }
 
-proc testHeapOwned(type t) {
+
+proc testHeap(type t) {
   var l = new heap(t);
 
   var x: t = new t(1);
 
   l.push(x);
   assert(l.size == 1);
 
-  var value = l.pop();
+  if !isOwnedClass(t) {
+    var value = l.top();
+  }
+  l.pop();
   assert(l.size == 0);
 
   if isUnmanagedClass(t) {
     delete x;
   }
-
```

Comment:
The above changes gave me a pause like "we should not have to create a separate version for `borrowed`." However I do not see how we would avoid that.

Reply:
Yes, actually the earlier approach was leading to, effectively, `new borrowed C`, which is currently deprecated.